### PR TITLE
catch2 3.0.1

### PR DESCRIPTION
**Pull request by Porkepix**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  test.cpp:2:10: fatal error: 'catch2/catch.hpp' file not found
  #include <catch2/catch.hpp>
           ^~~~~~~~~~~~~~~~~~
  1 error generated.
  Error: catch2: failed
```

**Comment by carlocab** [comment]
https://github.com/catchorg/Catch2/blob/v3.0.1/docs/migrate-v2-to-v3.md#migrating-from-v2-to-v3

**Comment by ChrisThrasher** [comment]
CI is failing because C++11 is used but Catch2 v3 requires C++14. Change `-std=c++11` to `-std=c++14` near the bottom of Formula/catch2.rb.

**Comment by ChrisThrasher** [comment]
I think we need to add `-lCatch2Main` to link to Catch2 with `main()` compiled in.

File: Formula/catch2.rb [hunk 1]
```diff
@@ -33,7 +32,7 @@ def install
         }
       }
     EOS
-    system ENV.cxx, "test.cpp", "-std=c++11", "-o", "test"
+    system ENV.cxx, "test.cpp", "-std=c++14", "-o", "test"
```
Comment (SMillerDev) [inline Formula/catch2.rb:35]:
```suggestion
    system ENV.cxx, "test.cpp", "-lCatch2Main", "-std=c++14", "-o", "test"
```

**Comment by ChrisThrasher** [comment]
I fixed more CI issues in https://github.com/Homebrew/homebrew-core/pull/102192. I figured out how to get Catch2 building but now downstream projects are breaking.

**Comment by ChrisThrasher** [comment]
https://github.com/Homebrew/homebrew-core/pull/102192 got approved so this PR can be closed.

**Pull request closed without merging**